// @fenderdigital/react-logo/lib/Logo.js
'use strict';

exports.__esModule = true;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _logoMap = require('./logo-map');

var _logoMap2 = _interopRequireDefault(_logoMap);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var getLogo = function getLogo(appName, type) {
  if (_logoMap2.default[appName]) {
    var imgType = _logoMap2.default[appName][type] ? type : 'primary';
    return {
      app: appName + '-' + imgType,
      image: _logoMap2.default[appName][type] || _logoMap2.default[appName]['primary']
    };
  }
  return {
    app: 'com-primary',
    image: _logoMap2.default.com.primary
  };
};

var Logo = function Logo(_ref) {
  var appName = _ref.appName,
      type = _ref.type;

  var _getLogo = getLogo(appName, type),
      app = _getLogo.app,
      image = _getLogo.image;

  return _react2.default.createElement(
    'div',
    { className: 'logo dib', 'data-id': 'logo-' + app },
    appName ? _react2.default.createElement('img', { alt: 'logo-' + app, src: image }) : null
  );
};

Logo.propTypes = process.env.NODE_ENV !== "production" ? {
  appName: _propTypes2.default.string,
  type: _propTypes2.default.string
} : {};

Logo.defaultProps = {
  appName: null,
  type: 'primary'
};

exports.default = Logo;
module.exports = exports['default'];
